api/v1alpha1/storageclaim_types.go

```diff
@@ -52,10 +54,10 @@ type StorageClientNamespacedName struct {
 // StorageClaimSpec defines the desired state of StorageClaim
 type StorageClaimSpec struct {
 	//+kubebuilder:validation:Enum=block;sharedfile
-	Type             string                       `json:"type"`
-	EncryptionMethod string                       `json:"encryptionMethod,omitempty"`
-	StorageProfile   string                       `json:"storageProfile,omitempty"`
-	StorageClient    *StorageClientNamespacedName `json:"storageClient"`
+	Type              string  `json:"type"`
+	EncryptionMethod  string  `json:"encryptionMethod,omitempty"`
+	StorageProfile    string  `json:"storageProfile,omitempty"`
+	StorageClientName *string `json:"storageClient"`
```

Comment:
Strings can be empty and nil. There should be a really good reason having an API field be defined as a pointer to a string. I don't see any reason to do it here.

Reply:
I didn't change any, https://ibm.enterprise.slack.com/archives/C065Q8HCSSC/p1709787993883729
tl:dr; version is, if client field is empty we pick existing client, however this field can't be empty due to crd cel rule and didn't touch due to reason mentioned in above thread.

Reply:
But you did. Before the change, it was an embedded structure. That meant that in order to make it optional we needed to use a pointer to a struct. But now it is a plain string, and strings can be nil (or empty) without us needing to add a supporting pointer.

Reply:
string can't be nil, it has an empty value, isn't it?

Reply:
Yes, you are correct (regarding string and ni values), it was a mistake on my side. But regardless, a string empty value has the same meaning in this case. If the user does not provide a claim name or provide an empty claim name we should react in exactly the same way.  And, assuming we defined the type correctly, omitted claim names will get deserialized into an empty string.

Reply:
> ommited claim names
- that's the interesting part, as per cel rule on CRD there can never be a ommited claim name
- we weren't creating claims per client earlier however we are doing that now. Given that, should we provide an option to have empty client name in claim which wasn't possible upto now?

Reply:
done.